modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/metadata/MetadataNegativeTestCase.java

```diff
@@ -83,22 +86,23 @@ public void processorDoesNotExist() throws Exception {
     componentId = new ProcessorId(CONTENT_AND_OUTPUT_METADATA_WITH_KEY_ID, notValidIndex);
     MetadataResult<ComponentMetadataDescriptor> metadata = metadataManager.getMetadata(componentId, PERSON_METADATA_KEY);
 
-    assertFailure(metadata, format(PROCESSOR_DOES_NOT_EXIST, notValidIndex), UNKNOWN, IndexOutOfBoundsException.class.getName());
+    assertFailure(metadata, format(PROCESSOR_DOES_NOT_EXIST, notValidIndex), COMPONENT_NOT_FOUND,
+                  InvalidComponentIdException.class.getName());
   }
 
   @Test
   public void failToGetMetadataFromNonExistingSource() throws IOException {
     final SourceId notExistingSource = new SourceId(FLOW_WITHOUT_SOURCE);
     final MetadataResult<MetadataKeysContainer> metadataKeysResult = metadataManager.getMetadataKeys(notExistingSource);
 
-    assertFailure(metadataKeysResult, SOURCE_DOES_NOT_EXIST, UNKNOWN, InvalidComponentIdException.class.getName());
+    assertFailure(metadataKeysResult, SOURCE_DOES_NOT_EXIST, COMPONENT_NOT_FOUND, InvalidComponentIdException.class.getName());
   }
 
   @Test
   public void processorIsNotEntityMetadataProvider() throws Exception {
     componentId = new ProcessorId(LOGGER_FLOW, FIRST_PROCESSOR_INDEX);
     MetadataResult<TypeMetadataDescriptor> metadata = metadataManager.getEntityMetadata(componentId, PERSON_METADATA_KEY);
-    assertFailure(metadata, NOT_A_METADATA_PROVIDER, UNKNOWN, InvalidComponentIdException.class.getName());
+    assertFailure(metadata, NOT_A_METADATA_PROVIDER, COMPONENT_NOT_FOUND, InvalidComponentIdException.class.getName());
```

Comment:
The logger does exist, but does not provide dynamic metadata. The FailureCode is different.